core/cas-server-core-services-registry/src/main/java/org/apereo/cas/services/AbstractServicesManager.java

```diff
@@ -98,11 +98,11 @@ public RegisteredService findServiceBy(final Service service) {
 
         val candidates = getCandidateServicesToMatch(service.getId());
         var foundService = configurationContext.getRegisteredServiceLocators()
-            .stream()
-            .map(locator -> locator.locate(candidates, service))
-            .filter(Objects::nonNull)
-            .findFirst()
-            .orElse(null);
+                .stream()
+                .map(locator -> locator.locate(candidates, service))
+                .filter(candidate -> candidate != null && validateRegisteredService(candidate) != null)
```

Comment:
Null checking here is unnecessary. The follow-up call does a null check already.

Reply:
Previous code filtered through `Objects::nonNull`, didn't want to rock the boat too much here. I'll take your word for it and remove.